ui/packages/catalog-ui-search/src/main/webapp/component/content-title/content-title.view.js

```diff
@@ -16,47 +16,63 @@
 
     return Marionette.LayoutView.extend({
         setDefaultModel: function(){
             this.model = store.get('content');
         },
-        regions: {
-            unsavedIndicator: '.title-saved'
+        behaviors() {
+            const view = this;
+            return {
+                region: {
+                    regions: [
+                        {
+                            selector: '.title-saved',
+                            view() {
+                                if (view.model.get('currentWorkspace')) {
+                                    return UnsavedIndicatorView;
+                                }   
+                                return undefined;
+                            },
+                            viewOptions() {
+                                return {
+                                    model: view.model.get('currentWorkspace')
+                                };
+                            },
+                            shouldRegionUpdate(currentView) {
+                                if (currentView && currentView.model !== view.model.get('currentWorkspace')) {
+                                    return true;
+                                }
+                                return false;
+                            }
+                        }
+                    ]
+                }
+            }
         },
-        events: {
-            'change input': 'updateWorkspaceName',
-            'keyup input': 'updateWorkspaceName',
-            'keydown input': 'updateWorkspaceName'
+        template(data) {
+            return (
+                <React.Fragment key={data.currentWorkspace.id}>
+                    <input placeholder="Workspace Title" value={data.currentWorkspace.title}
+                     data-help="This is the title of the workspace you are currently in.
+                    If you have permission, you can click here to start editing the title." onChange={this.updateWorkspaceName.bind(this)}/>
+                    <pre className="title-display">{data.currentWorkspace.title}</pre>
+                    <div className="title-saved"></div>
+                </React.Fragment>
+            )
         },
-        template: template,
         tagName: CustomElements.register('content-title'),
         initialize: function (options) {
             if (options.model === undefined){
                 this.setDefaultModel();
             }
-            this.listenTo(this.model, 'change:currentWorkspace', this.updateIndicator);
-            this.listenTo(this.model, 'change:currentWorkspace', this.handleSaved);
-        },
-        onBeforeShow: function(){
-            this.updateIndicator();
-        },
-        handleSaved: function(){
-            var currentWorkspace = this.model.get('currentWorkspace');
-            this.$el.toggleClass('is-saved', currentWorkspace ? currentWorkspace.isSaved() : false);
-        },
-        updateIndicator: function(workspace){
```

Comment:
This got moved into the region behavior logic.  It's a more advanced usage that takes advantage of the `shouldRegionUpdate` functionality of the region behaviors.  Basically, on subsequent renders this function is called by the behavior to see if the view should be destroyed and reshown.  As a result, all we do when the currentWorkspace changes now is call render.

Notice that all of the definitions for this particular region behavior utilize the function definition rather than a static definition.  As a result it can change across subsequent renders (if and only if `shouldRegionUpdate` is implemented).